src/components/Configure/actions/onSaveWriteCreateInstallation.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  api, CreateInstallationOperationRequest,
+  CreateInstallationRequestConfig,
+  HydratedRevision,
+  Installation,
+} from '../../../services/api';
+import { ConfigureState } from '../types';
+
+/**
+   * gets write objects from hydratedRevision
+   * @param hydratedRevision
+   * @param objectName
+   * @returns
+   */
+const getWriteObjectsFromHydratedRevision = (
+  hydratedRevision: HydratedRevision,
+) => {
+  const writeAction = hydratedRevision.content.write;
+  return writeAction?.objects;
+};
+
+type WriteObject = {
+  objectName: string;
+};
+
+type WriteObjects = {
+  [objectName: string]: WriteObject;
+};
+
+/**
+ * example type
+ * "objects":
+ *  {
+    objects: {
+      account: {
+        objectName: 'account',
+      },
+      contact: {
+        objectName: 'contact',
+      },
+    },
+  }
+ * @param writeObjects
+ * @param configureState
+ * @returns
+ */
+const generateConfigWriteObjects = (configureState: ConfigureState) => {
+  const configWriteObjects: WriteObjects = {}; // any is listed type in generated spec
+  const configStateWriteObjects = configureState.write?.writeObjects;
+  if (configStateWriteObjects) {
+    configStateWriteObjects.forEach((configStateWriteObject) => {
+      // object exists in config form
+      if (configStateWriteObject.objectName) {
```

Comment:
can you put `configStateWriteObject.objectName` in a variable? since it's being used 3 times.